pkg/scheduler/scheduler.go

```diff
@@ -469,8 +469,10 @@ func (sched *Scheduler) addAllEventHandlers() {
 					klog.ErrorDepth(5, fmt.Sprintf("failed to parse labelSelector in Subscription %s: %v", key, err))
 					continue
 				}
-				if !selector.Matches(labels.Set(newMcls.Labels)) && !selector.Matches(labels.Set(oldMcls.Labels)) {
-					continue
+				if !selector.Matches(labels.Set(newMcls.Labels)) {
+					if oldMcls == nil || !selector.Matches(labels.Set(oldMcls.Labels)) {
```

Comment:
`if oldMcls != nil && !selector.Matches(labels.Set(oldMcls.Labels))` will be more readable.

Reply:
👌， done